x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx

```diff
@@ -502,6 +505,71 @@ const StatefulEventsViewerComponent: React.FC<EventsViewerProps & PropsFromRedux
     [totalCountMinusDeleted, unit]
   );
 
+  const rowHeightsOptions: EuiDataGridRowHeightsOptions | undefined = useMemo(() => {
+    if (tableView === 'eventRenderedView') {
+      return {
+        defaultHeight: 'auto' as const,
+      };
+    }
+    return undefined;
+  }, [tableView]);
+
+  const dataTable = useMemo(
+    () => (
+      <DataTableComponent
```

Comment:
I am not sure there's any benefit in doing this _useMemo_ of the _DataTableComponent_ component, it is already `React.memo`ed. The only prop that would makes sense to memo is the `pagination` prop which is a new object every time, maybe we could do the _useMemo_ for the `pagination` prop and render this component directly at the end.